nexus/src/external_api/params.rs

```diff
@@ -182,6 +182,20 @@ pub struct DiskCreate {
     pub size: ByteCount,
 }
 
+impl DiskCreate {
+    pub fn block_size(&self) -> u64 {
+        512
+    }
+
+    pub fn extent_size(&self) -> u64 {
+        1 << 20
+    }
+
+    pub fn extent_count(&self) -> u64 {
+        (self.size.to_bytes() + self.extent_size() - 1) / self.extent_size()
```

Comment:
I wonder if this `extent_count` is going to match what Crucible has.   Maybe it does not need to?

Reply:
When a client requests a volume, it requests:
- Size of the volume

On the other hand, the parameters I need to supply to the Crucible Agent client API are:
- Block Size
- Extent Count
- Blocks per Extent
 
I need to have some mapping from that "size" to the Crucible Agent input, and figured that:
- Block size is arbitrary
- Blocks per extent is arbitrary
- Extent count needs to be large enough to fit the size of the volume

I did make the assumption that "extent count" = "enough extents to contiguously hold the client volume size, rounded up to the nearest extent". If this *isn't* the mapping between "volume size" and "extent count", how can I provide a value that'll be big enough to contain the volume through Crucible's API?

Reply:
Your assumption is correct, and I think the issues (#181 and #183 in crucible) you created should help us to figure out how to proceed from here.  You (and this code) is correct in that you are required to provide these values to the Crucible Agent, I just wanted to be sure that the extent_count here matches what you send to Crucible.